app/javascript/dashboard/routes/dashboard/helpcenter/composables/usePortal.js

```diff
@@ -0,0 +1,52 @@
+import { useAccount } from 'dashboard/composables/useAccount';
+import { frontendURL } from 'dashboard/helper/URLHelper';
+import allLocales from 'shared/constants/locales.js';
+import { useRoute } from 'dashboard/composables/route';
+import { computed } from 'vue';
+
+/**
+ * @typedef {Object} PortalComposable
+ * @property {import('vue').ComputedRef<number>} accountId - The current account ID.
+ * @property {import('vue').ComputedRef<string>} portalSlug - The slug of the current portal.
+ * @property {import('vue').ComputedRef<string>} locale - The current locale code.
+ * @property {function(number): string} articleUrl - A function to generate the URL for an article.
+ * @property {function(string): string} localeName - A function to get the localized name of a locale.
+ */
+
+/**
+ * A composable for managing portal-related data and utilities.
+ * @returns {PortalComposable} An object containing portal-related properties and functions.
+ */
+export const usePortal = () => {
+  const { accountId } = useAccount();
+  const portalSlug = computed(() => useRoute().params.portalSlug);
+  const locale = computed(() => useRoute().params.locale);
```

Comment:
Let's store the result of useRoute() in a `const` instead of calling it multiple times
```suggestion
  const route = useRoute();
  const { accountId } = useAccount();
  const portalSlug = computed(() => route.params.portalSlug);
  const locale = computed(() => route.params.locale);
```